test/e2eprovider/vault/vault.go

```diff
@@ -0,0 +1,177 @@
+package e2eprovider
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+)
+
+// E2eVault is a fake Vault for end-to-end testing.
+type E2eVault struct {
```

Comment:
Let's make it more generic and call it `E2EStore`. WDYT?

Reply:
removed vault